pubsublite/consumer_test.go

```diff
@@ -60,6 +69,283 @@ func TestNewConsumer(t *testing.T) {
 		config.Delivery = 100
 		_, err := NewConsumer(context.Background(), config)
 		assert.Error(t, err)
-		assert.EqualError(t, err, "pubsublite: invalid consumer config: pubsublite: delivery is not valid")
+		assert.EqualError(t, err, "pubsublite: invalid consumer config: delivery is not valid")
 	})
 }
+
+func TestConsumerConsume(t *testing.T) {
+	type subscriptionPartition struct {
+		Subscription string
+		Partition    int64
+	}
+
+	ch := make(chan apmqueue.Record)
+	server, config := newConsumerService(t)
+	topic1SubscriptionPath := "projects/project/locations/region-1/subscriptions/name_space-topic1+consumer_name"
+	topic2SubscriptionPath := "projects/project/locations/region-1/subscriptions/name_space-topic2+consumer_name"
+
+	server.startSession = func(session *subscriberSession, _ *pubsublitepb.SeekRequest) {
+		switch session.subscription {
+		case topic1SubscriptionPath:
+		case topic2SubscriptionPath:
+		default:
+			panic("unexpected subscription: " + session.subscription)
```

Comment:
```suggestion
			t.Fatal("unexpected subscription: " + session.subscription)
```

Reply:
This code is running in a different goroutine - t.Fatal should only be called from the main test goroutine.